R/theme.r

```diff
@@ -538,6 +532,61 @@ calc_element <- function(element, theme, verbose = FALSE) {
   Reduce(combine_elements, parents, theme[[element]])
 }
 
+#' Merge a parent element into a child element
+#'
+#' This is a generic and element classes must provide an implementation of this
+#' method
+#'
+#' @param new The child element in the theme hierarchy
+#'
+#' @param old The parent element in the theme hierarchy
+#'
+#' @return A modified version of \code{new} updated with the properties of
+#' \code{old}
+#'
+#' @keywords internal
+#' @export
+#'
+#' @examples
+#'
+#' new <- element_text(colour = "red")
+#' old <- element_text(colour = "blue", size = 10)
+#'
+#' # Adopt size but ignore colour
+#' merge_element(new, old)
+#'
+merge_element <- function(new, old) {
+  UseMethod("merge_element")
+}
+#' @rdname merge_element
+#' @export
+merge_element.default <- function(new, old) {
+  stop("No method for merging ", class(new), " into ", class(old), call. = FALSE)
+}
+merge_element_default <- function(new, old) {
```

Comment:
My reason for this setup is to avoid ppl overriding the behaviour of the base elements if just `merge_element.element` is provided there is nothing stopping users from providing `neege_element.text` etc